packages/capabilities/src/filecoin/dealer.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Filecoin Dealer Capabilities
+ *
+ * These can be imported directly with:
+ * ```js
+ * import * as Dealer from '@web3-storage/capabilities/filecoin/dealer'
+ * ```
+ *
+ * @module
+ */
+
+import { capability, Schema, ok } from '@ucanto/validator'
+import { PieceLink } from './lib.js'
+import { equalWith, checkLink, and } from '../utils.js'
+
+/**
+ * Capability allowing an Aggregator to request an aggregate to be added to a
+ * deal with a Storage Provider.
+ */
+export const aggregateOffer = capability({
+  can: 'aggregate/offer',
+  /**
+   * DID of an authorized Storefront.
+   */
+  with: Schema.did(),
+  nb: Schema.struct({
+    /**
+     * Commitment proof for the aggregate being offered.
+     */
+    aggregate: PieceLink,
+    /**
+     * CID of the DAG-CBOR encoded block with offer details.
+     * Service will queue given offer to be validated and handled.
+     */
+    pieces: Schema.link(),
+  }),
+  derives: (claim, from) => {
+    return (
+      and(equalWith(claim, from)) ||
+      and(checkLink(claim.nb.aggregate, from.nb.aggregate, 'nb.aggregate')) ||
+      and(checkLink(claim.nb.pieces, from.nb.pieces, 'nb.pieces')) ||
+      ok({})
+    )
+  },
+})
+
+/**
+ * Capability that allows a Dealer to signal an aggregate has been accepted
+ * for inclusion in a Filecoin deal.
+ */
+export const aggregateAccept = capability({
+  can: 'aggregate/accept',
+  /**
+   * did:key identifier of the broker authority where offer is made available.
+   */
+  with: Schema.did(),
+  nb: Schema.struct({
+    /**
+     * Commitment proof for the aggregate being offered.
+     */
+    aggregate: PieceLink,
+    /**
+     * CID of the DAG-CBOR encoded block with offer details.
+     * Service will queue given offer to be validated and handled.
+     */
+    pieces: Schema.link(),
```

Comment:
While working on the CRON that will read pending deals to issue `aggregate/accept` for the ones already on chain, I questioned this need.

`pieces` is quite big list, for which we dag encode it, and inline block with the block. We then store it.

But, we also store it on `aggregate/offer`. This list is actually stored both in the UCAN Stream CAR bucket, as well as in the store where Spade reads from.

This brings the question, should we don't care about putting `pieces` here? There will only be one aggregate with that CID, and we already store its content in 2 different places. Sending it to self invoke `aggregate/accept` from Dealer seems useless and requires us to write a new copy of this list on our UCAN Stream CAR bucket. Moreover, makes implementation more complex and expensive. After we track deals and see that a deal is now on chain, we only have the aggregateCid in the deals Table. So, for then invoking `aggregate/accept` we would need to read the list from one of the already written places and put it there.

We could do a middle term as well, and store the `invocationCid` + `taskCid` (of `aggregate/offer` on the `deal-store` and add them here instead).